File: Backend/src/main/java/com/smartamenities/backend/dto/RouteRequest.java
package com.smartamenities.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for receiving POST /route request body.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RouteRequest {
    private String destination;
    private String currentLocation;
    private boolean accessibilityOn;
    private String destinationAmenityId;
    private Double currentLatitude;
    private Double currentLongitude;
}

File: Mobile/app/src/main/java/com/smartamenities/app/model/RouteRequest.kt
package com.smartamenities.app.model

data class RouteRequest(
    val destination: String,
    val currentLocation: String,
    val accessibilityOn: Boolean,
    val destinationAmenityId: String? = null,
    val sessionSeed: String? = null,
    val currentLatitude: Double? = null,
    val currentLongitude: Double? = null
)

